src/Umbraco.New.Cms.Infrastructure/Factories/Installer/DatabaseSettingsFactory.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Options;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Mapping;
+using Umbraco.Cms.Infrastructure.Persistence;
+using Umbraco.Extensions;
+using Umbraco.New.Cms.Core.Factories;
+using Umbraco.New.Cms.Core.Models.Installer;
+
+namespace Umbraco.New.Cms.Infrastructure.Factories.Installer;
+
+public class DatabaseSettingsFactory : IDatabaseSettingsFactory
+{
+    private readonly IEnumerable<IDatabaseProviderMetadata> _databaseProviderMetadata;
+    private readonly IOptionsMonitor<ConnectionStrings> _connectionStrings;
+    private readonly IUmbracoMapper _mapper;
+
+    public DatabaseSettingsFactory(
+        IEnumerable<IDatabaseProviderMetadata> databaseProviderMetadata,
+        IOptionsMonitor<ConnectionStrings> connectionStrings,
+        IUmbracoMapper mapper)
+    {
+        _databaseProviderMetadata = databaseProviderMetadata;
+        _connectionStrings = connectionStrings;
+        _mapper = mapper;
+    }
+
+    /// <inheritdoc/>
+    public ICollection<DatabaseSettingsModel> GetDatabaseSettings()
+    {
+        ConnectionStrings? connectionString = _connectionStrings.CurrentValue;
+
+        // If the connection string is configured we just return the configured provider.
+        if (connectionString.IsConnectionStringConfigured())
+        {
+            var providerName = connectionString.ProviderName;
+            if (providerName is null)
```

Comment:
Consider testing for empty/whitespace string here also - `if(providerName.IsNullOrWhiteSpace())`

Reply:
I think we can completely remove this check because `IsConnectionStringConfigured` does it: 

```C#
    public static bool IsConnectionStringConfigured(this ConnectionStrings connectionString)
        => connectionString != null &&
        !string.IsNullOrWhiteSpace(connectionString.ConnectionString) &&
        !string.IsNullOrWhiteSpace(connectionString.ProviderName);
```